_extensions/odoo/__init__.py

```diff
@@ -28,6 +28,8 @@ def setup(app):
     app.connect('html-page-context', update_meta)
 
 def update_meta(app, pagename, templatename, context, doctree):
+    if not context.get('meta'):
+        context['meta'] = {}
```

Comment:
Can `context['meta']` take another falsy value than `{}` ?

Reply:
I did not experienced any. Either the key does not exist or it is a dictionary.

Reply:
Then I may be missing something but why do you need to set its default value if it's directly followed by a setdefault ?

Reply:
Because setdefault returns the value of the key. 

When context['meta'] is not defined, it is set to None.
```
context = {'meta': None}
meta = context.setdefault('meta')
```
meta is equal to None and therefore, meta.setdefault('banner', app.config.odoo_cover_default) crashes. You have the same problem with the .get method:
```
context = {'meta': None}
var = context.get('meta', {})
```
var is still equal to None.